src/o365/spo/commands/file/file-get.ts

```diff
@@ -0,0 +1,277 @@
+      })
+      .then((res: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        if (this.verbose) {
+          cmd.log(`Retrieving file from site ${args.options.webUrl}...`);
+        }
+
+        let requestUrl: string;
+        let options: string;
+
+        if (args.options.id) {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileById('${encodeURIComponent(args.options.id)}')`;
+        }
+        else {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(args.options.url)}')`;
+        }
+
+        if (args.options.asListItem) {
+          options = '?$expand=ListItemAllFields';
+        }
+        else if (!args.options.asListItem && !args.options.asFile && !args.options.asString) {
+          options = '';
+        }
+        else {
+          options = '/$value';
+        }
+
+        const requestOptions: any = {
+          url: requestUrl + options,
+          method: 'GET',
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'accept': 'application/json;odata=nometadata'
+          }),
+          encoding: null, // Set encoding to null, otherwise binary data will be encoded to utf8 and binary data is corrupt 
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((file: string): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(file);
+          cmd.log('');
+        }
+        
+        if (args.options.asString) {
+          cmd.log(file);
+        }
+        else if (args.options.asListItem) {
+          let fileProperties: FileProperties = JSON.parse(JSON.stringify(file));
+          cmd.log(fileProperties.ListItemAllFields)
+        }
+        else if(args.options.asFile) {
+          this.writeFile(file, args.options.path, args.options.fileName)
+          cmd.log(`File ${args.options.fileName} saved to path ${args.options.path}`);
+        }
+        else {
+          let fileProperties: FileProperties = JSON.parse(JSON.stringify(file));
+          cmd.log(fileProperties);
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  private writeFile(fileContent: string, filePath: string, fileName: string): void {
+    let fullPath: string = path.join(filePath, fileName);
+
+    fs.writeFileSync(fullPath, fileContent);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-w, --webUrl <webUrl>',
+        description: 'The URL of the site where the folder from which to retrieve files is located'
```

Comment:
This doesn't seem quite right ;)

Reply:
Changed description in order to reflect the one from the specification.